src/client/csfle/options.rs

```diff
@@ -0,0 +1,140 @@
+use std::collections::HashMap;
+
+use bson::Array;
+use mongocrypt::ctx::KmsProvider;
+use typed_builder::TypedBuilder;
+
+use crate::{
+    bson::{doc, Bson, Document},
+    client::options::TlsOptions,
+    error::{Error, Result},
+};
+
+/// Options related to automatic encryption.
+///
+/// Automatic encryption is an enterprise only feature that only applies to operations on a
+/// collection. Automatic encryption is not supported for operations on a database or view, and
+/// operations that are not bypassed will result in error (see [libmongocrypt: Auto Encryption
+/// Allow-List](
+/// https://github.com/mongodb/specifications/blob/master/source/client-side-encryption/client-side-encryption.rst#libmongocrypt-auto-encryption-allow-list
+/// )). To bypass automatic encryption for all operations, set bypassAutoEncryption=true in
+/// AutoEncryptionOpts.
+#[derive(Debug, Default, TypedBuilder, Clone)]
+#[non_exhaustive]
+#[builder(field_defaults(setter(into)))]
+pub struct AutoEncryptionOpts {
+    /// Used for data key queries.  Will default to an internal client if not set.
+    #[builder(default)]
+    pub key_vault_client: Option<crate::Client>,
+    /// A collection that contains all data keys used for encryption and decryption (aka the key
+    /// vault collection).
+    pub key_vault_namespace: String,
```

Comment:
we have a `Namespace` type, should we use that rather than just a string?

Reply:
Yup, done.